crates/swc_ecma_loader/src/resolvers/node.rs

```diff
@@ -83,14 +83,23 @@ enum StringOrBool {
 pub struct NodeModulesResolver {
     target_env: TargetEnv,
     alias: AHashMap<String, String>,
+    resolve_symlinks: bool,
 }
 
 static EXTENSIONS: &[&str] = &["ts", "tsx", "js", "jsx", "json", "node"];
 
 impl NodeModulesResolver {
     /// Create a node modules resolver for the target runtime environment.
-    pub fn new(target_env: TargetEnv, alias: AHashMap<String, String>) -> Self {
-        Self { target_env, alias }
+    pub fn new(
+        target_env: TargetEnv,
+        alias: AHashMap<String, String>,
+        resolve_symlinks: bool,
+    ) -> Self {
+        Self {
+            target_env,
+            alias,
+            resolve_symlinks,
+        }
     }
 
     fn wrap(&self, path: Option<PathBuf>) -> Result<FileName, Error> {
```

Comment:
Isn't it better to canonicalize here?

Reply:
(Assuming the entry is not an symlink)

Reply:
If we canonicalize at here, `base` on next resolving would use the canonicalized path, and it will reduce duplicated call to `base.canonicalize`.